pkg/pool-stable/test/StableMath.test.ts

```diff
@@ -27,18 +26,33 @@ describe('StableMath', function () {
   });
 
   context('invariant', () => {
-    context('two tokens', () => {
-      it('returns invariant', async () => {
-        const amp = bn(100);
-        const balances = [fp(10), fp(12)];
-
-        const result = await mock.invariant(amp.mul(AMP_PRECISION), balances, true);
-        const expectedInvariant = calculateInvariant(balances, amp);
-
-        expectEqualWithError(result, expectedInvariant, MAX_RELATIVE_ERROR);
-      });
+    async function checkInvariant(balances: BigNumber[], amp: number): Promise<void> {
+      const ampParameter = bn(amp).mul(AMP_PRECISION);
+
+      const result = await mock.invariant(ampParameter, balances, true);
+      const expectedInvariant = calculateInvariant(balances, ampParameter);
+
+      expectEqualWithError(result, expectedInvariant, MAX_RELATIVE_ERROR);
+    }
+
+    context('check over a range of inputs', () => {
+      for (let numTokens = 2; numTokens <= 5; numTokens++) {
+        let balances = Array(numTokens).fill(300);
+        balances = balances.map((b) => (random(100) > 50 ? b + random(50) : b - random(50)));
+
+        it(`computes the invariant for ${numTokens} tokens`, async () => {
+          for (let amp = 100; amp <= 5000; amp += 100) {
+            await checkInvariant(
+              balances.map((b) => fp(b)),
+              amp
+            );
```

Comment:
```suggestion
        const balances = Array(numTokens).fill(300)
          .map((b) => (random(100) > 50 ? b + random(50) : b - random(50)))
          .map(fp);

        it(`computes the invariant for ${numTokens} tokens`, async () => {
          for (let amp = 100; amp <= 5000; amp += 100) {
            await checkInvariant(balances, amp);
```

Reply:
I find this slightly more straightforward, with all of the balances mutation in a single place